backend/src/payments/payment.controller.ts

```diff
@@ -83,75 +83,57 @@ export class PaymentController {
   async payment_return(@Req() req, @Res() res) {
     try {
       const response = req.query.RETURN_CODE;
-      if (response === this.BamboraReturnCodes.SUCCESS) {
-        const orderNumber = req.query.ORDER_NUMBER;
-        const payment = await this.paymentService.getPaymentByOrderNumber(
-          orderNumber,
-        );
-        if (!payment) {
-          console.error(
-            `Payment details not available in the system. Payment Order: ${
-              payment.order_number
-            }`,
-          );
-          return res.redirect(`${APP_REDIRECT_URL}?status=1`);
-        }
-
-        const reservation_details = await this.reservationService.findOneById(
-          payment.reservation_id,
-        );
-        // Checking if the payment has already been done.
-        if (payment.payment_status) {
-          console.error(
-            `This payment was already processed!. Payment Order: ${
-              payment.order_number
-            }`,
-          );
-          return res.redirect(
-            `${APP_REDIRECT_URL}?status=2&event_id=${reservation_details.id}`,
-          );
-        } else {
-          // Updating payment as completed.
-          payment.payment_status = true;
-          await this.paymentService.updatePaymentStatus(
-            payment.order_number,
-            true,
-          );
-        }
-        await this.reservationService.updatePaymentStatus(
-          payment.reservation_id,
-          true,
-        );
-
-        const smsResponse = await this.paymentService.sendSmsToUser(
-          payment.reservation_id,
-          payment.amount,
-        );
-        if (smsResponse) {
-          res.redirect(
-            `${APP_REDIRECT_URL}?orderNumber=${orderNumber}&amount=${
-              payment.amount
-            }&status=0&event_id=${reservation_details.id}`,
-          );
-        } else {
-          res.redirect(
-            `${APP_REDIRECT_URL}?orderNumber=${orderNumber}&amount=${
-              payment.amount
-            }&status=5&event_id=${reservation_details.id}`,
-          );
-        }
-      } else {
+      const orderNumber = req.query.ORDER_NUMBER;
+      const payment = await this.paymentService.getPaymentByOrderNumber(
+        orderNumber,
+      );
+      if (response !== this.BamboraReturnCodes.SUCCESS) {
+        //delete reservation since payment failed
+        await this.reservationService.deleteReservation(payment.reservation_id);
         console.error(
           `Payment failed with error code: ${response}. Please try again later`,
```

Comment:
Yes. its a good to have. I think its needed for the project as a whole at various levels and can be introduced later. Not at this point when we are making bug fixes over the budget. While we do the next feature additions this can be added.